cosmos/providers/dbt/core/operators/virtualenv.py

```diff
@@ -0,0 +1,146 @@
+from __future__ import annotations
+
+import logging
+from pathlib import Path
+from tempfile import TemporaryDirectory
+
+from airflow.compat.functools import cached_property
+from airflow.utils.context import Context
+from airflow.utils.python_virtualenv import prepare_virtualenv
+
+from cosmos.providers.dbt.core.operators.local import (
+    DbtDocsLocalOperator,
+    DbtLocalBaseOperator,
+    DbtLSLocalOperator,
+    DbtRunLocalOperator,
+    DbtRunOperationLocalOperator,
+    DbtSeedLocalOperator,
+    DbtSnapshotLocalOperator,
+    DbtTestLocalOperator,
+)
+
+logger = logging.getLogger(__name__)
+
+
+PY_INTERPRETER = "python3"
+
+
+class DbtVirtualenvBaseOperator(DbtLocalBaseOperator):
+    """
+    Executes a dbt core cli command within a Python Virtual Environment, that is created before running the DBT command
+    and deleted at the end of the operator execution.
+
+    :param py_requirements: If defined, creates a virtual environment with the specified dependencies. Example:
+           ["dbt-postgres==1.5.0"]
+    :param py_system_site_packages: Whether or not all the Python packages from the Airflow instance will be accessible
+           within the virtual environment (if py_requirements argument is specified).
+           Avoid using unless the DBT job requires it.
```

Comment:
```suggestion
    Executes a dbt core cli command within a Python Virtual Environment, that is created before running the dbt command
    and deleted at the end of the operator execution.

    :param py_requirements: If defined, creates a virtual environment with the specified dependencies. Example:
           ["dbt-postgres==1.5.0"]
    :param py_system_site_packages: Whether or not all the Python packages from the Airflow instance will be accessible
           within the virtual environment (if py_requirements argument is specified).
           Avoid using unless the dbt job requires it.
```
Nit for consistency of "dbt" in the docstring. There are a few other places "DBT" is used in the changes, but up to you to implement whichever formatting. I don't have strong opinions on the format -- I know dbt folks do 🙂 -- but consistency of choice is good I think.

Reply:
Thanks, @josh-fell , it was a copy & paste issue - just fixed in the `virtualenv.py` module. I'm adding a ticket to make this consistent across the project.

Reply:
This is the follow up ticket: #306